internal/xdscache/v3/listener.go

```diff
@@ -477,23 +497,26 @@ func (v *listenerVisitor) visit(vertex dag.Vertex) {
 				vh.DownstreamValidation,
 				alpnProtos...)
 
+			builder := envoy_v3.HTTPConnectionManagerBuilder().
+				DefaultFilters().
+				RouteConfigName(ENVOY_FALLBACK_ROUTECONFIG).
+				MetricsPrefix(ENVOY_HTTPS_LISTENER).
+				AccessLoggers(v.ListenerConfig.newSecureAccessLog()).
+				RequestTimeout(v.ListenerConfig.RequestTimeout).
+				ConnectionIdleTimeout(v.ListenerConfig.ConnectionIdleTimeout).
+				StreamIdleTimeout(v.ListenerConfig.StreamIdleTimeout).
+				DelayedCloseTimeout(v.ListenerConfig.DelayedCloseTimeout).
+				MaxConnectionDuration(v.ListenerConfig.MaxConnectionDuration).
+				ConnectionShutdownGracePeriod(v.ListenerConfig.ConnectionShutdownGracePeriod).
+				AllowChunkedLength(v.ListenerConfig.AllowChunkedLength).
+				NumTrustedHops(v.ListenerConfig.XffNumTrustedHops)
+
+			if rlc := v.RateLimitConfig; rlc != nil {
```

Comment:
nit: Could this be collapsed so we don't need to add the if (here and above)?

e.g. `envoy_v3.GlobalRateLimitFilter` could take just a `RateLimitConfig` and do the `nil` check there? and `builder.AddFilter` does nothing if it gets `nil`?

Reply:
Yeah good idea, that'd clean things up a bit; though as-is that would create an import cycle so need to re-jigger things a bit.